mm2src/derives/enum_from/src/from_stringify.rs

```diff
@@ -0,0 +1,127 @@
+use quote::{quote, ToTokens};
+use syn::__private::TokenStream2;
+use syn::spanned::Spanned;
+use syn::Variant;
+
+#[derive(Clone)]
+pub struct MapEnumNestedData {
+    variant_ident: Ident,
+    nested_meta: Vec<syn::NestedMeta>,
+    inner_ident: Option<Ident>,
+}
+
+#[derive(Debug)]
+pub(crate) enum InnerIdentTypes {
+    String,
+    Named,
+    None,
+}
+
+pub(crate) fn get_inner_ident_type(ident: Option<Ident>) -> InnerIdentTypes {
+    if let Some(ident) = ident {
+        return match ident.to_string().as_str() {
+            "String" => InnerIdentTypes::String,
+            _ => InnerIdentTypes::Named,
+        };
+    };
+
+    InnerIdentTypes::None
+}
+
+pub(crate) fn get_attributes(variants: Variant) -> Result<MapEnumNestedData, CompileError> {
+    let variant_ident = &variants.ident;
+    let fields = &variants.fields;
+
+    for attribute in variants.attrs {
+        if let Ok(meta) = attribute.parse_meta() {
+            return match meta {
+                syn::Meta::List(syn::MetaList { nested, .. }) => {
+                    return Ok(MapEnumNestedData {
+                        variant_ident: variant_ident.to_owned(),
+                        nested_meta: nested.into_iter().collect(),
+                        inner_ident: get_variant_unnamed_ident(fields.to_owned()),
+                    });
+                },
+                _ => Err(CompileError::expected_enum_from_stringify(
+                    MacroAttr::FromStringify,
+                    &attribute.tokens.to_string(),
+                )),
+            };
+        };
+    }
+    Err(CompileError::expected_enum_from_stringify(
+        MacroAttr::FromStringify,
+        &variant_ident.to_string(),
+    ))
+}
+
+fn get_variant_unnamed_ident(fields: syn::Fields) -> Option<Ident> {
+    if let syn::Fields::Unnamed(syn::FieldsUnnamed { unnamed, .. }) = fields {
+        if let Some(field) = unnamed.iter().next() {
+            if let Some(syn::Type::Path(type_path, ..)) = field.ty.next().cloned() {
+                let type_path = type_path.path.segments.iter().next().cloned()?.ident;
+                return Some(type_path);
+            };
+        };
+    }
+    None
+}
+
+pub(crate) fn map_enum_data_from_variant(variant: &Variant) -> Result<Vec<MapEnumNestedData>, CompileError> {
+    let mut meta_vec = vec![];
+    let attr = get_attributes(variant.to_owned())?;
+    for nested_meta in attr.nested_meta {
+        let variant_ident = attr.variant_ident.to_owned();
+        meta_vec.push(MapEnumNestedData {
+            variant_ident,
+            nested_meta: vec![nested_meta],
+            inner_ident: attr.inner_ident.clone(),
+        });
+    }
+    Ok(meta_vec)
+}
+
+fn parse_inner_ident(attr: &TokenStream) -> Result<Ident, CompileError> {
+    let ident_to_impl_from = attr.to_string();
+
+    let strip_prefix = ident_to_impl_from.strip_prefix('\"').unwrap();
+    let strip_suffix = strip_prefix.strip_suffix('\"').unwrap();
```

Comment:
I don't know where exactly it panics, but I got
```
error: proc-macro derive panicked
    --> mm2src/coins/lp_coins.rs:1445:33
     |
1445 | #[derive(Clone, Debug, Display, EnumFromStringify, EnumFromTrait, Serialize, SerializeErrorType, PartialEq)]
     |                                 ^^^^^^^^^^^^^^^^^
     |
     = help: message: called `Option::unwrap()` on a `None` value
```
after these changes:
```diff
diff --git a/mm2src/coins/lp_coins.rs b/mm2src/coins/lp_coins.rs
index e7bc9fd0c..225aed051 100644
--- a/mm2src/coins/lp_coins.rs
+++ b/mm2src/coins/lp_coins.rs
@@ -38,7 +38,7 @@ use base58::FromBase58Error;
 use common::{calc_total_pages, now_ms, ten, HttpStatusCode};
 use crypto::{Bip32Error, CryptoCtx, DerivationPath, HwRpcError, WithHwRpcError};
 use derive_more::Display;
-use enum_from::EnumFromTrait;
+use enum_from::{EnumFromStringify, EnumFromTrait};
 use futures::compat::Future01CompatExt;
 use futures::lock::Mutex as AsyncMutex;
 use futures::{FutureExt, TryFutureExt};
@@ -1442,7 +1442,7 @@ impl DelegationError {
     }
 }
 
-#[derive(Clone, Debug, Display, EnumFromTrait, Serialize, SerializeErrorType, PartialEq)]
+#[derive(Clone, Debug, Display, EnumFromStringify, EnumFromTrait, Serialize, SerializeErrorType, PartialEq)]
 #[serde(tag = "error_type", content = "error_data")]
 pub enum WithdrawError {
     #[display(
```

I believe it's related to the following edge cases:
1. You should check if the enum variant has `#[from_stringify("Inner")]` attribute, and only then try to parse and implement `From<Inner>` trait.
2. Also the proc-macro should work fine if there are other attributes:
```rust
enum WithdrawError {
  #[from_trait(WithInternal::internal)]
  #[from_stringify("NumConversError", "UnexpectedDerivationMethod")]
  #[display(fmt = "Internal error: {}", _0)]
  InternalError(String),
}
```

To reach these, please consider repeating the code that is already written for `EnumFromInner` and `EnumFromTrait`:
https://github.com/KomodoPlatform/atomicDEX-API/blob/6c90c28a29723dad6c48296bed3093686188eeb2/mm2src/derives/enum_from/src/from_trait.rs#L22-L27
https://github.com/KomodoPlatform/atomicDEX-API/blob/6c90c28a29723dad6c48296bed3093686188eeb2/mm2src/derives/enum_from/src/from_trait.rs#L84-L85

Also try to avoid using `unwrap` and `expect` functions as they may panic without any useful information.

Reply:
thanks fixed this

Reply:
Please consider this
> Also try to avoid using unwrap and expect functions as they may panic without any useful information.